NewForce-Cohort-5/tabloidfullstack-actual-spring
# Hb delete tag tkt15

# Description

An admin user should be able to see and click on the Tag Management link in the menu to get the list of tags.
The user will see a list of tags that are in alphabetical order.
There is an edit option. These options will not work at this time.
The user can create new and delete options.
The user can click delete and is directed to a confirm delete page. 
The user can then click the button confirm delete and delete the tag and go back to the tag list. Where they will see the tag has been deleted. 
The other option user can also click go back to list and reject deletion. They will see the tag is still there. 

In the backend, I have added the delete method to the Tag Repository  and delete method on the TagController.   

The front end, I added a modal to the tag card to give the user a confirmation message.  I also added a deleteTag fetch call in the provider.  
 
There are no permissions for this ticket but will be included in future tickets. Assume all users are admins.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.


# Testing Instructions

1. After verifying that you have our repo cloned down into a local repository, `git fetch -a` in the main branch to retrieve the named `hb-deleteTag-tkt15` that I have pushed up
2. In Terminal `git checkout hb-deleteTag-tkt15`  to switch to my branch and in the terminal type `Start TabloidMVC.sln` to open in Visual Studios 2022.
3. To get to VS code, type `cd Tabloid `and then `cd Client` and type` code .`
4. If you have not seeded your database with this data, seed your data from SQLQuery link :
5. You will need to create the table first and then add the seed data. Choose files in the following order:   

-    [01_Tabloid_Create_DB.sql](https://github.com/NewForce-Cohort-5/tabloidfullstack-actual-spring/blob/main/SQL/01_Tabloid_Create_DB.sql)
-    [02_Tabloid_Seed_Data.sql](https://github.com/NewForce-Cohort-5/tabloidfullstack-actual-spring/blob/main/SQL/02_Tabloid_Seed_Data.sql)


6. Now that you have your seed data, we can test.
7. Click the `solid green debugger play button` and you will arrive at the swagger page. We don’t need to do anything here. This just confirms the server is running. 
8. In your terminal, you should be in the client folder where `Package.json` is located. You want to type `npm start` to start your react app.  
9. You will arrive at a login page where you will use the email login as [foo@bar.com](mailto:foo@bar.com) and password test.
10. Next go to `Tag Management` in the menu. 
11. Please test at least 2 times by switching between menu items links and logging in and out
12. Click `create new` and add a tag called test.
13. You will click on the `delete button` on the test tag and you will be directed to a confirmation pop up.  
14. To test, first click `back to list`  to reject the deletion and confirm the tag is still there.  
15. Second, click `delete`  again on the test tag and then click confirm delete.
16. You should be routed back to the list and the tag should no longer be listed.   
  


# Checklist:
